app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt

```diff
@@ -281,6 +290,34 @@ class DefaultSessionControlController(
         )
     }
 
+    override fun handleRenameTopSiteClicked(topSite: TopSite) {
+        if (topSite.id != null) {
+            activity.let {
+                val customLayout =
+                        LayoutInflater.from(it).inflate(R.layout.rename_top_site_dialog, null)
+                val topSiteLabelEditText: EditText =
+                        customLayout.findViewById(R.id.top_site_title)
+                topSiteLabelEditText.setText(topSite.title)
+
+                AlertDialog.Builder(it).setTitle(R.string.rename_top_site)
+                    .setView(customLayout).setPositiveButton(android.R.string.ok) { _, _ ->
+                        val newTitle = topSiteLabelEditText.text.toString()
+                        if (newTitle.isNotBlank()) {
+                            viewLifecycleScope.launch(Dispatchers.IO) {
+                                with(activity.components.useCases.topSitesUseCase) {
+                                    renameTopSites(topSite, newTitle)
+                                }
+                            }
+                        }
+                    }.setNegativeButton(android.R.string.cancel, null)
```

Comment:
I think you will need to reuse the strings as defined in https://github.com/mozilla-mobile/fenix/pull/16239/files#diff-5e01f7d37a66e4ca03deefc205d8e7008661cdd0284a05aaba1858e6b7bf9103 because of localization.